lib/console_web/controllers/v1/flow_controller.ex

```diff
@@ -0,0 +1,122 @@
+    current_organization = conn.assigns.current_organization
+
+    case Channels.get_channel(current_organization, id) do
+      nil ->
+        {:error, :not_found, "Integration not found"}
+      %Channel{} = channel ->
+        cond do
+          Map.has_key?(params, "device_id") and Map.has_key?(params, "label_id") ->
+            {:error, :bad_request, "Flow must not contain both device_id and label_id"}
+          Map.has_key?(params, "device_id") ->
+            device = Devices.get_device!(current_organization, Map.get(params, "device_id"))
+            function =
+              if Map.has_key?(params, "function_id") do
+                Functions.get_function!(current_organization, Map.get(params, "function_id"))
+              else
+                nil
+              end
+
+            if function != nil do
+              with {:ok, _} <- Flows.create_flow(%{ device_id: device.id, channel_id: channel.id, function_id: function.id, organization_id: current_organization.id }) do
+                broadcast_router_update_devices([device.id])
+
+                conn
+                |> send_resp(:ok, "Flow with integration created successfully")
+              end
+            else
+              with {:ok, _} <- Flows.create_flow(%{ device_id: device.id, channel_id: channel.id, organization_id: current_organization.id }) do
+                broadcast_router_update_devices([device.id])
+
+                conn
+                |> send_resp(:ok, "Flow with integration created successfully")
+              end
+            end
+          Map.has_key?(params, "label_id") ->
+            label = Labels.get_label!(current_organization, Map.get(params, "label_id"))
+            function =
+              if Map.has_key?(params, "function_id") do
+                Functions.get_function!(current_organization, Map.get(params, "function_id"))
+              else
+                nil
+              end
+
+            if function != nil do
+              with {:ok, _} <- Flows.create_flow(%{ label_id: label.id, channel_id: channel.id, function_id: function.id, organization_id: current_organization.id }) do
+                device_ids =
+                  Devices.get_devices_for_label(label.id)
+                  |> Enum.map(fn d -> d.id end)
+
+                broadcast_router_update_devices(device_ids)
+
+                conn
+                |> send_resp(:ok, "Flow with integration created successfully")
+              end
+            else
+              with {:ok, _} <- Flows.create_flow(%{ label_id: label.id, channel_id: channel.id, organization_id: current_organization.id }) do
+                device_ids =
+                  Devices.get_devices_for_label(label.id)
+                  |> Enum.map(fn d -> d.id end)
+
+                broadcast_router_update_devices(device_ids)
+
+                conn
+                |> send_resp(:ok, "Flow with integration created successfully")
+              end
+            end
```

Comment:
could this be simpler by conditionally adding `label_id`, `device_id`, and `function_id` to a variable, say we call it `params`, and then just calling `Flows.create_flow(params)` in one same `with` clause? may be easier to read and less same lines of code

Reply:
let me try, my brain wasn't super on point earlier